unfoldingWord/translationCore
# node csv - #2434

#### This pull request addresses:

- Refactored CSV Export to address a bunch of csv issues. 
- Updated to use node-csv library.
- Added tests for most of the stack.
- Improved error handling to catch more issues.
- Should handle bogus files in project data folder where it didn't before.
- Handles Autographa data where it didn't before.
- Looks up proper groupNames where many times they weren't found.
- Fixes issue where checkdata was exported twice both with partial data. 
- Fixes issues where csv exported  files were corrupted and malformed.

#### How to test this pull request:

- `npm install`
- `npm test`
- Go to projects page and exporting to CSV should work on all projects now.
- If you find a project that doesn't succeed document it so we can add it to the tests and fix it.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/unfoldingword-dev/translationcore/2553)
